depthai_nodes/ml/parsers/parser.py

```diff
@@ -0,0 +1,59 @@
+from typing import Dict, List, Union
+
+import depthai as dai
+
+
+class Parser(dai.node.ThreadedHostNode):
+    """Base class for all parsers."""
+
+    def __init__(self):
+        super().__init__()
+        self.input = self.createInput()
+        self.out = self.createOutput()
+
+        self.head_configs: Dict = {}
+
+    def build(self, head_metadata: Union[List, Dict], name: str = ""):
+        """Initial build method for all parsers."""
+
+        if isinstance(head_metadata, list):
+            if len(head_metadata) == 0:
+                raise ValueError("No heads parsed from archive")
+            if len(head_metadata) > 1 and name == "":
+                raise ValueError("Multiple heads detected, please specify head name")
```

Comment:
Can we make this even smarter: 
If we are initializing ClassificationParser and only one head has `ClassificationParser˙ set as `parser` then we can automatically pick that one even if we get a whole list of all heads passed in.
Something like this:
```
current_parser = self.__class__.__name__ # this should return "ClassificationParser"
all_heads_parsers = [h.parser for h in head_metadata]
num_matches = all_heads_parsers.cound(current_parser)
if num_matches == 1:
    # good, take that one
else:
    # need head name
```

Reply:
Sounds good, I'll add it.